website/src/controllers/api-v3/groups.js

```diff
@@ -292,9 +292,20 @@ api.joinGroup = {
     group.memberCount += 1;
 
     let promises = [group.save(), user.save()];
+    let partyAchievements;
 
     if (group.type === 'party' && inviter) {
       promises.push(User.update({_id: inviter}, {$inc: {'items.quests.basilist': 1}}).exec()); // Reward inviter
+      if (group.memberCount > 3) {
+        partyAchievements = {$set: {'achievements.partyUp': true, 'achievements.partyOn': true}};
+      } else if (group.memberCount > 1) {
+        partyAchievements = {$set: {'achievements.partyUp': true}};
+      }
+      if (partyAchievements) {
+        _.forEach(group.members, (member) => {
```

Comment:
I think you'll want to do something like `promises.push(User.update({'party.id': group._id}, partyAchievements, {multi: true}).exec());`, check the syntax because I'm not sure it's correct